x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawler_overview.test.tsx

```diff
@@ -24,29 +79,68 @@ describe('CrawlerOverview', () => {
 
   const mockValues = {
     dataLoading: false,
-    domains: [],
+    domains,
+    crawlRequests,
   };
 
-  let wrapper: ShallowWrapper;
-
   beforeEach(() => {
     jest.clearAllMocks();
-    setMockValues(mockValues);
     setMockActions(mockActions);
-    wrapper = shallow(<CrawlerOverview />);
   });
 
   it('calls fetchCrawlerData on page load', () => {
+    setMockValues(mockValues);
+
+    shallow(<CrawlerOverview />);
+
     expect(mockActions.fetchCrawlerData).toHaveBeenCalledTimes(1);
   });
 
-  it('renders', () => {
-    expect(wrapper.find(DomainsTable)).toHaveLength(1);
+  it('hides the domain and crawl request tables when there are no domains, and no crawl requests', () => {
+    setMockValues({ ...mockValues, domains: [], crawlRequests: [] });
+
+    const wrapper = shallow(<CrawlerOverview />);
+
+    expect(wrapper.find(AddDomainForm)).toHaveLength(1);
+    expect(wrapper.find(AddDomainFormSubmitButton)).toHaveLength(1);
+    expect(wrapper.find(AddDomainFlyout)).toHaveLength(0);
+    expect(wrapper.find(DomainsTable)).toHaveLength(0);
+    expect(wrapper.find(CrawlRequestsTable)).toHaveLength(0);
+  });
+
+  it('shows the domain and the crawl request tables when there are domains, but no crawl requests', () => {
+    setMockValues({ ...mockValues, crawlRequests: [] });
 
-    // TODO test for CrawlRequestsTable after it is built in a future PR
+    const wrapper = shallow(<CrawlerOverview />);
 
+    expect(wrapper.find(AddDomainForm)).toHaveLength(0);
+    expect(wrapper.find(AddDomainFormSubmitButton)).toHaveLength(0);
     expect(wrapper.find(AddDomainFlyout)).toHaveLength(1);
+    expect(wrapper.find(DomainsTable)).toHaveLength(1);
+    expect(wrapper.find(CrawlRequestsTable)).toHaveLength(1);
+  });
+
+  it('hides the domain table and shows the crawl request tables when there are crawl requests but no domains', () => {
+    setMockValues({ ...mockValues, domains: [] });
+
+    const wrapper = shallow(<CrawlerOverview />);
+
+    expect(wrapper.find(AddDomainForm)).toHaveLength(1);
+    expect(wrapper.find(AddDomainFormSubmitButton)).toHaveLength(1);
+    expect(wrapper.find(AddDomainFlyout)).toHaveLength(0);
+    expect(wrapper.find(DomainsTable)).toHaveLength(0);
+    expect(wrapper.find(CrawlRequestsTable)).toHaveLength(1);
+  });
 
-    // TODO test for empty state after it is built in a future PR
+  it('shows the domain and the crawl request tableswhen there are crawl requests and domains', () => {
```

Comment:
Super small test name nit
```suggestion
  it('shows the domain and the crawl request tables when there are crawl requests and domains', () => {
```
These are super clear branches / easy to read test paths also, thank you for the great tests as always! 💯